source/Halibut/Transport/Streams/CallUnderlyingAsyncMethodsStream.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Runtime.Remoting;
+using System;
+
+namespace Halibut.Transport.Streams
+{
+    class CallUnderlyingAsyncMethodsStream : AsyncStream
```

Comment:
To ensure we covered all the methods that `NetworkTimeoutStream` would call, we copied that class, and altered appropriately.